src/encoder/encoderfdkaac.cpp

```diff
@@ -0,0 +1,429 @@
+#include "encoder/encoderfdkaac.h"
+
+#include <QDir>
+#include <QStandardPaths>
+#include <QString>
+#include <QStringList>
+
+#include "engine/sidechain/enginesidechain.h"
+#include "recording/defs_recording.h"
+#include "util/logger.h"
+#include "util/sample.h"
+
+namespace {
+// recommended in encoder documentation, section 2.4.1
+const int kOutBufferBits = 6144;
+const mixxx::Logger kLogger("EncoderFdkAac");
+} // namespace
+
+EncoderFdkAac::EncoderFdkAac(EncoderCallback* pCallback)
+        : aacEncOpen(nullptr),
+          aacEncClose(nullptr),
+          aacEncEncode(nullptr),
+          aacEncInfo(nullptr),
+          aacEncoder_SetParam(nullptr),
+          m_aacAot(AOT_AAC_LC),
+          m_bitrate(0),
+          m_channels(0),
+          m_samplerate(0),
+          m_pCallback(pCallback),
+          m_library(nullptr),
+          m_pInputFifo(nullptr),
+          m_pFifoChunkBuffer(nullptr),
+          m_readRequired(0),
+          m_aacEnc(),
+          m_pAacDataBuffer(nullptr),
+          m_aacInfo(),
+          m_hasSbr(false) {
+    // Load shared library
+    // Code import from encodermp3.cpp
+    QStringList libnames;
+#ifdef __LINUX__
+    libnames << "fdk-aac";
+    libnames << "libfdk-aac.so.2";
+    libnames << "libfdk-aac.so.1";
+#elif __WINDOWS__
+    // Give top priority to libfdk-aac copied
+    // into Mixxx's installation folder
+    libnames << "libfdk-aac-2.dll";
+    libnames << "libfdk-aac-1.dll";
+
+    // Fallback and user-friendly method: use libfdk-aac
+    // provided with B.U.T.T installed in
+    // a standard location
+    QString buttFdkAacPath = buttWindowsFdkAac();
+    if (!buttFdkAacPath.isNull()) {
```

Comment:
```suggestion
    if (!buttFdkAacPath.isEmpty()) {
```
https://doc.qt.io/qt-5/qstring.html#isNull
> Qt makes a distinction between null strings and empty strings for historical reasons. For most applications, what matters is whether or not a string contains any data, and this can be determined using the isEmpty() function.